packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1946,6 +1949,11 @@ function commitRootImpl(root, renderPriorityLevel) {
       nextEffect.nextEffect = null;
       nextEffect = nextNextEffect;
     }
+    // Clear the nextEffect pointers of an alternates
```

Comment:
The comment above says:

>If we have passive effects, we'll clear this in flushPassiveEffects.	

So I think we need to fix it there too. Otherwise it'll only work if you don't have passive effects. If you do, we'll keep adding things to the list and not clear them.

Reply:
I'm unsure of where you mean here. Adding the above logic into `flushPassiveEffects` causes issues. Maybe you know more here and can help me?